### PR TITLE
Propagate a lazy proxy of the storage model

### PR DESCRIPTION
**Pull request by poncovka**
The storage model for the partitioning modules should be always created lazily.
When we reset the partitioning, the new model shouldn't be created until we try
to work with it. Then we create a new copy of the storage model, hide disks that
are not selected and possibly initialize empty disks.

There is a problem with modules that need to be able to work with the same copy
of the storage model as the partitioning modules. At this moment, it is only the
device tree module. The suggested solution is to propagate a lazy proxy of the
storage model. It will not trigger the creation of the copy until we try to
access the attributes of the storage model.

Basically, the device tree module always gets the storage model on demand from
the storage property of the partitioning module.

Related: rhbz#1868577

### CODE REVIEW

**Review by jkonecny12 — COMMENTED** [review]
Looks good to me. It just feel fragile. Will this work even if some thread will be slower or similar situations?

**Review by jkonecny12 — APPROVED** [review]
Thanks a lot for the explanation. You are free to merge this.
